src/contract-ui/tabs/edit-extensions/components/ExtensionForm.tsx

```diff
@@ -0,0 +1,406 @@
+        contract,
+        chain: contract.chain,
+        client: contract.client,
+        account,
+        extensionName: watch("extensionContract"),
+        publisherAddress: watch("publisherAddress"),
+        version: watch("version"),
+      });
+
+      const txResult = await sendTransaction({
+        transaction: installTransaction,
+        account,
+      });
+
+      await waitForReceipt(txResult);
+    },
+  });
+
+  const onSubmit = async () => {
+    try {
+      await installMutation.mutateAsync();
+
+      props.refetchExtensions();
+
+      toast({
+        variant: "solid",
+        position: "bottom",
+        title: "Extension installed successfully",
+        status: "success",
+        duration: 5000,
+        isClosable: true,
+      });
+
+      // clear form
+      reset({
+        publisherAddress: "",
+        extensionContract: "",
+        version: "latest",
+      });
+    } catch (error) {
+      toast({
+        variant: "solid",
+        position: "bottom",
+        title: "Failed to install extension",
+        status: "error",
+        duration: 5000,
+        isClosable: true,
+      });
+
+      console.error("Error during installation:", error);
+    }
+  };
+
+  const extensionContractInputProps = register("extensionContract", {
+    required: "Extension name is required",
+  });
+
+  const selectedExtension = publishedContractsQuery.data?.find(
+    (x) => x.id === watch("extensionContract"),
+  );
+
+  // Get core contract bytecode
+  const coreContractByteCodeQuery = useQuery({
+    queryKey: ["getBytecode", contract.address],
+    queryFn: async () => {
+      const coreImplementation = await resolveImplementation(contract);
+      return coreImplementation.bytecode;
+    },
+    retry: false,
+    // 30 minutes
+    staleTime: 1000 * 60 * 30,
+  });
+
+  // Get Installed extension bytecodes
+  const installedExtensionBytecodesQuery = useQuery({
+    queryKey: [
+      "installedExtensionBytecodes",
+      contract.address,
+      props.installedExtensions.data,
+    ],
+    queryFn: async () => {
+      const extensionAddress = props.installedExtensions.data;
+      if (!extensionAddress) {
+        return [];
+      }
+
+      return Promise.all(
+        extensionAddress.map(async (address) => {
+          const result = await resolveImplementation({
+            client: thirdwebClient,
+            address,
+            chain: contract.chain,
+          });
+
+          if (!result) {
+            throw new Error("Failed to fetch bytecode for extension");
+          }
+
+          return result.bytecode;
+        }),
+      );
+    },
+    enabled: !!props.installedExtensions.data,
+    retry: false,
+    // 30 minutes
+    staleTime: 1000 * 60 * 30,
+  });
+
+  // check if selected extension is compatible with the core contract and installed extensions
+  const isExtensionCompatibleQuery = useQuery({
+    queryKey: [
+      "isExtensionCompatible",
+      contract,
```

Comment:
contract is a deep object, probably want contract.address & contract.chainId or something here

Reply:
added